enhancements/builds/disable-builder-sa.md

```diff
@@ -0,0 +1,362 @@
+
+
+
+#### Variation and form factor considerations [optional]
+
+- Adminstrators for standalone OCP will be able to modify the `Build`
+  configuration resource.
+- For hosted control planes, we currently do not allow the `Build` subsystem to
+  be modified for hosted clusters. This could change in the future. [1]
+- For Microshift/edge clusters, the `Build` APIs are not enabled [2]. This is
+  less likely to change due to the smaller form factor of Microshift.
+- For clusters installed without the `Build` capability [3], the system build
+  configuration CRD is not installed and its default instance is not created.
+  The cluster admin must first enable the `Build` capability, at which point
+  the configuration CRD and `cluster` instance are created.
+
+[1] https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1beta1.ClusterConfiguration
+[2] https://github.com/openshift/microshift/blob/4.15.0-rc.5-202402022103.p0/docs/contributor/enabled_apis.md
+[3] https://docs.openshift.com/container-platform/4.14/post_installation_configuration/enabling-cluster-capabilities.html#setting_additional_enabled_capabilities_enabling-cluster-capabilities
+
+
+### API Extensions
+
+Modified extension:
+- apiVersion: `config.openshift.io/v1`
+- kind: `Build`
+
+This proposal will update an existing cluster configuration CRD. The new field
+`builderServiceAccount` should default to `Generate` on upgrade. OCM-O should
+also consider an empty value as `Generate` and update the spec to reflect the
+controller's interpretation. The following values are allowed:
+
+- `Generate`
+- `Disable`
+
+Other values should be considered invalid for the time being, and reverted to
+`Generate` by ocm-o if detected. Follow up features may add additional
+supported values (ex: `Remove`).
+
+
+### Implementation Details/Notes/Constraint
+
+Many aspects of of the Build system and its user experience assume the
+`builder` service account is created in every namespace. Disabling the auto
+generation of this account, its RBAC, and its push secret may cause builds to
+fail. To address this, builds should "fail fast" if the service account
+specified for the build does not exist at pod creation time.
+
+For platform teams that want to "bring their own builder" service account, OCP
+should provide detailed documentation describing the RBAC that is generated for
+the `builder` service account.
+
+
+#### Hypershift [optional]
+
+
+No specific considerations are needed for Hypershift at present, as the `Build`
+cluster configuration resources are not exposed through hosted control plane
+APIs. If these configurations are exposed in the future, Hypershift will need
+to ensure appropriate behavior of openshift-controller-manager for each "guest
+cluster." Refactoring decisions - such as running the build-related controllers
+in their own `Deployment` - may be considered at such time.
+
+
+### Risks and Mitigations
```

Comment:
Add here a risk which Michal described in https://github.com/openshift/enhancements/pull/1558/files#r1488230335